### PR TITLE
Add template engine example

### PR DESCRIPTION
**Pull request by simon-johansson**

### CODE REVIEW

File: ts/template-engine/encore.app [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+	"id":   "",
+	"lang": "typescript",
+	"build": {
+    "docker": {
+      "bundle_source": true
+    }
```
Comment (eandre) [inline ts/template-engine/encore.app:7]: Indentation here is off

File: ts/template-engine/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "Encore.ts Template Engine Example",
+  "license": "MPL-2.0",
+  "type": "module",
+  "devDependencies": {
+    "@types/ejs": "^3.1.5",
+    "@types/node": "^20.5.7",
+    "typescript": "^5.2.2"
+  },
+  "dependencies": {
+    "@types/consolidate": "^0.14.4",
```
Comment (eandre) [inline ts/template-engine/package.json:14]: Move to `devDependencies` presumably

File: ts/template-engine/template/template.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const inlineHTML = `
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="stylesheet" href="/public/styles.css" >
+  </head>
+  <body>
+    <h1>Static Inlnie HTML Example</h1>
```
Comment (eandre) [inline ts/template-engine/template/template.ts:42]:
```suggestion
    <h1>Static Inline HTML Example</h1>
```

File: ts/template-engine/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Template Engine with Encore.ts
+
+This example utilizes a Raw endpoint to handle serve views through a template engine. This example uses the [EJS](https://ejs.co/) for rendering views but the same concept could be used for a different template engine.
```
Comment (marcuskohlberg) [inline ts/template-engine/README.md:3]:
```suggestion
This example utilizes a Raw endpoint to serve views through a template engine. The example implementation uses [EJS](https://ejs.co/) for rendering views, but the same approach is also applicable for other template engines.
```

File: ts/template-engine/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+**Pages:**
+
+- <http://localhost:4000/person> - Serving a specific view with data
+- <http://localhost:4000/> - Serving a template based on the dynamic path
+- <http://localhost:4000/about/contact> - Serving a template located in a sub-folder
+- <http://localhost:4000/html> - Serving an inline HTML template
+
+Go to [http://localhost:4000](http://localhost:4000) in the browser to see the example frontend.
```
Comment (marcuskohlberg) [inline ts/template-engine/README.md:35]:
```suggestion
Open [http://localhost:4000](http://localhost:4000) in your browser to see the example frontend.
```

File: ts/template-engine/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+You can also access Encore's [local developer dashboard](https://encore.dev/docs/observability/dev-dash) on <http://localhost:9400/> to view traces, API documentation, and more.
```
Comment (marcuskohlberg) [inline ts/template-engine/README.md:37]:
```suggestion
You can also access Encore's [local development dashboard](https://encore.dev/docs/observability/dev-dash) on <http://localhost:9400/> to view traces, API documentation, and more.
```

**Comment by simon-johansson** [comment]
Thanks for the comments!

**Pull request merged**